Utils/Dataflow/095_datasetInfoAMI/amiDatasets.py

```diff
@@ -118,7 +121,13 @@ def amiPhysValues(data):
     - k_factor
     - cross_section_ref
     """
-    dataset = data['datasetname']
+    dataset = data.get('datasetname')
+    if dataset is None:
+        sys.stderr.write("(WARN) No datasetname found in data:%r\n" % data)
+        return False
+    elif not isinstance(dataset, (str, unicode)):
+        sys.stderr.write("(WARN) Datasetname is not a string:%r\n" % dataset)
+        return False
```

Comment:
Done: fec92d1.

Reply:
There\`s this difficulty with existing code refactoring/improvements: it is very tempting to just rewrite it all and say "code rewritten". But all the changes have to be done and logged step by step, so that it was easy to understand where things were just refactored to improve code readability/extendability/usability/..., and where something was changed in the actual code behaviour (reliability improved, bugs fixed, etc). For one simple reason: if something went wrong after this reworking, it may be anything: typo, copy-and-paste mistake, logical error, forgotten renaming... and it is better to have set of smaller commits where you know what deserves special attention -- then a single commit with the whole variety of (possible) mistakes.

The mentioned commit (fec92d1), as I see it, does the following things:
* changes one function behaviour (which indeed invites renaming and description update);
* introduces additional checks.

However the summary (and even the whole message) does not reallly explain what and why happened. "Rework" means "something changed", fine. But then the description simply contains a list of changes that gives no clue why it happened and why all these changes are gathered in one commit. I mean, what was the general goal of all these changes?

If split things into two commits, first one may still be named as "rework", but should be accompanied with some explaination -- why new version is better? Something like "I didn\`t like the function name" would also work (except, it doesn\`t explain why the func behaviour has to be changed); but it must explain why author decided change something and what he/she tried to achieve by this.

Reply:
Hm. I had spent some time pondering how to correctly handle this situation before making fec92d1, but I agree with your comment, which means that not all problems were solved with it. Reworked the commit into three ones (not two, to further separate change of behavior and stylistic fixes of the description):
12d5ee9
4f88409
bf15935

Reply:
Looks good now, thank you!